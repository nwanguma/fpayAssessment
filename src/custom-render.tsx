import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const Wrapper: React.FC = ({ children }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
