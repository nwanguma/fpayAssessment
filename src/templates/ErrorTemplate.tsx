import { Link } from "react-router-dom";

import logo from "../assets/images/logo.svg";

interface IProps {
  children: JSX.Element;
  image: string;
}

const ErrorTemplate: React.FC<IProps> = ({ children, image }) => {
  return (
    <div className="wrapper">
      <header className="header--payout w-[100%] md:w-[90%] mx-auto flex items-center py-6 justify-between">
        <Link to="/">
          <img src={logo} alt="fliqpay" />
        </Link>
      </header>
      <main className="container min-h-[80vh] min-w-full flex items-center">
        <div className="container flex flex-col-reverse  md:flex-row justify-center md:justify-between items-center max-w-[400px] md:text-left md:max-w-[80%] mx-auto">
          <div className="container mt-3 flex-item w-[90%] text-center md:text-left">
            {children}
          </div>
          <div className="container mx-auto md:mx-0 w-[250px] h-[150px] md:w-[350px] md:h-[250px]">
            <img className="w-[100%] h-[100%]" src={image} alt="error pic" />
          </div>
        </div>
      </main>
    </div>
  );
};

export default ErrorTemplate;
