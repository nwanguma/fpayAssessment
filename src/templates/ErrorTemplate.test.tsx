import { render, screen } from "../custom-render";
import sample from "../assets/images/503.png";

import ErrorTemplate from "./ErrorTemplate";

describe("ErrorTemplate", () => {
  test("<ErrorTemplate /> matches snapshot", () => {
    const component = render(
      <ErrorTemplate image={sample}>
        <div>Test</div>
      </ErrorTemplate>
    );

    expect(component.container).toMatchSnapshot();
  });

  test("It renders children", () => {
    const App = render(
      <ErrorTemplate image={sample}>
        <div>Test</div>
      </ErrorTemplate>
    );

    expect(screen.getByTestId("wrapper")).toBeInTheDocument();
  });

  test("It renders the correct link", () => {
    render(
      <ErrorTemplate image={sample}>
        <div>Test</div>
      </ErrorTemplate>
    );

    expect(screen.getByAltText("fliqpay").closest("a")).toHaveAttribute(
      "href",
      "/"
    );
  });
});

export {};
