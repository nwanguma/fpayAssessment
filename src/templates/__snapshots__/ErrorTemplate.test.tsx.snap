// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTemplate <ErrorTemplate /> matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="wrapper"
  >
    <header
      class="header--payout w-[100%] md:w-[90%] mx-auto flex items-center py-6 justify-between"
    >
      <a
        href="/"
      >
        <img
          alt="fliqpay"
          src="logo.svg"
        />
      </a>
    </header>
    <main
      class="container min-h-[80vh] min-w-full flex items-center"
    >
      <div
        class="container flex flex-col-reverse  md:flex-row justify-center md:justify-between items-center max-w-[400px] md:text-left md:max-w-[80%] mx-auto"
      >
        <div
          class="container mt-3 flex-item w-[90%] text-center md:text-left"
        >
          <div>
            Test
          </div>
        </div>
        <div
          class="container mx-auto md:mx-0 w-[250px] h-[150px] md:w-[350px] md:h-[250px]"
        >
          <img
            alt="error pic"
            class="w-[100%] h-[100%]"
            src="503.png"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
